graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/fetchers/admin/DeploySchemaFetcherBase.java

```diff
@@ -100,7 +106,8 @@ protected DeploySchemaResponseDto get(
       SchemaSource newSource = schemaSourceDao.insert(keyspaceName, input);
       schemaSourceDao.purgeOldVersions(keyspaceName);
       response.setVersion(newSource.getVersion());
-      // TODO update SchemaFirstCache from here instead of letting it reload from the DB
+      graphqlCache.getGraphQLAndUpdateCache(
```

Comment:
One of the motivations behind this ticket was to avoid computing the GraphQL schema twice. We do it in this class at line 86:
```java
      ProcessedSchema processedSchema =
          new SchemaProcessor(authorizationService, dataStoreFactory, false)
              .process(input, keyspace);
      ...
      queries =
          CassandraMigrator.forDeployment(migrationStrategy)
              .compute(processedSchema.getMappingModel(), keyspace);
```

But when we call the cache here, the same work will be done a second time (see `DmlGraphHolder.buildGraphql`).

So we should expose a different cache method to directly put the already computed result:
```java
graphqlCache.put(newSource, processedSchema.getGraphql(), authenticationSubject.customProperties()));
```
That method will simply compute the decorated keyspace and call `dmlGraphqls.put()`.
We also need a new `DmlGraphqlHolder` constructor that takes a `GraphQL` directly, but that should be a pretty straightforward change.

(PS: this can race if someone accesses the schema after we've written the `SchemaSource` but before we've completed the cache update, but that doesn't matter, the worst that can happen is that the schema gets computed twice)

Reply:
Done, one question regarding possible race:
https://github.com/stargate/stargate/pull/950#discussion_r633419019